docs/custom-code/add-custom-dto.md

```diff
@@ -0,0 +1,243 @@
+---
+id: add-custom-dto
+title: How to add a custom DTO
+sidebar_label: Add a custom DTO
+slug: /custom-code/custom-dto
+---
+
+# How to add a custom DTO
+
+## General
+A DTO (Data Access Object) is an object that defines how the data will be sent over the network. 
+DTOs can be defined by TypeScript interfaces or by classes. Though, it is preferable to use classes, as TypeScript interfaces are removed during the the phase of translating TypeScript to JavaScript, and can't be referred an at runtime. However, classes are part of the JavaScript ES6 standard and are therefore preserved as real entities in the compiled JavaScript.
+
+The DTOs in Amplication are generated based on the relations between the entities and Prisma schema and types.
+Sometimes you have use-cases where you need DTOs supported by Prisma but not generated by Amplication yet.
```

Comment:
this is just one use case and should not be used as the main intro... 

Better using something like "In addition to the default DTOs that are generated by Amplication, you may need to create a new custom DTO or to extend an existing one...". 

For example,